### DISCUSSION TITLE
:bike: :house: Reformat the CHANGELOG to one line per PR

### DISCUSSION THREAD

**Discussion by lnicola** [georust/gdal#488] [General]
The changelog currently looks like this:

![image](https://github.com/georust/gdal/assets/308347/7a361e5a-2d3e-4a7b-b554-2abc642bd654)

It takes a lot of space in both directions and it feels like half of the PRs that touch it get the indentation wrong.

I propose we switch to a more standard layout, like:

## Breaking

 - Replaced `TryFrom<&[(&str, &str); N]>` for CslStringList with `impl FromIterator<CslStringListEntry> for CslStringList`, `impl FromIterator<String> for CslStringList` and `impl<'a> FromIterator<&'a str> for CslStringList`
 - `CslStringListIterator` returns a `CslStringListEntry` instead of `(String, String)` in order to differentiate between `key=value` entries and `flag`s
 - `CslStringList::fetch_name_value` returns `Option<String>` instead of `Result<Option<String>>`, better reflecting the semantics of the GDAL API
 - `SpatialRef::set_axis_mapping_strategy` now takes `&mut self` (#461)
 - `Dataset::raster_count` now returns `usize` and `Dataset::rasterband` takes `usize` instead of `isize` (#434)
 - `ExtendedDataType` no longer implements `Clone`, `PartialEq` and `Eq` (#451)
 - `LayerIterator`, `LayerOptions` and `Transaction` were moved to `crate::vector` (#447)

## Added

 - support for Digital Elevation Model raster processing: `aspect`, `color_relief`, `hillshade`, `roughness`, `slope`, `terrain_ruggedness_index`, `topographic_position_index` (#456)
 - pre-built bindings for GDAL 3.8 (#466)
 - `{Display|FromStr}` for `ResampleAlg` and `ResampleAlg::iter` (#462)
 - `impl Extend<CslStringListEntry> for CslStringList` and `CslStringList::merge` (#457)
 - `CslStringList::get_field`, `CslStringList::find_string`, `CslStringList::partial_find_string`, `CslStringList::find_string_case_sensitive`, `CslStringList::into_ptr`, `CslStringList::add_name_value` (#455)
 - `MajorObject::gdal_object_ptr` and `Dataset::c_dataset` are no longer `unsafe` (#447)
 - histogram calculation (#468)

## Fixed
 - the `gdal-sys` build script no longer fails on development GDAL versions (#439)

If there's support for this, I can do the translation myself.

**Comment by jdroenner** [comment]
that would be great :)

**Comment by metasim** [comment]
@lnicola Thanks for starting this conversation!

I think the biggest problem was the fact that the PR link was indented at 3 spaces, a level that various editors would interpret differently. In your proposal I like that:

1. You're just using the PR number and not the full link
2. You're putting it at the end of the line.

Grouping by change type is a bonus improvement. :)

**Comment by metasim** [comment]
@lnicola Any ideas on how we get from here to there?

**Reply by lnicola** [reply]
Tomorrow! #514 was a bit too much :weary:.
